src/client/mod.rs

```diff
@@ -941,45 +943,59 @@ impl Client {
 /// // ensure a valid token is in fact valid:
 /// assert!(validate_token("Mjg4NzYwMjQxMzYzODc3ODg4.C_ikow.j3VupLBuE1QWZng3TMGH0z_UAwg").is_ok());
 ///
-/// // "cat" isn't a valid token:
-/// assert!(validate_token("cat").is_err());
-///
-/// // tokens must have three parts, separated by periods (this is still
-/// // actually an invalid token):
-/// assert!(validate_token("aaa.abcdefgh.bbb").is_ok());
-///
-/// // the second part must be _at least_ 6 characters long:
-/// assert!(validate_token("a.abcdef.b").is_ok());
-/// assert!(validate_token("a.abcde.b").is_err());
+/// // helpful to prevent typos
+/// assert!(validate_token("Njg4NzYwMjQxMzYzODc3ODg4.C_ikow.j3VupLBuE1QWZng3TMGH0z_UAwg").is_err());
 /// ```
 ///
 /// # Errors
 ///
 /// Returns a [`ClientError::InvalidToken`] when one of the above checks fail.
 /// The type of failure is not specified.
 pub fn validate_token(token: impl AsRef<str>) -> Result<()> {
-    let token = token.as_ref();
+    let (user_id, generation_time) =
+        parse_token(token.as_ref()).ok_or(Error::Client(ClientError::InvalidToken))?;
 
-    if token.is_empty() {
+    dbg!(user_id, generation_time);
+
+    // Check if timestamps are in a sensible range
+    if user_id.created_at().year() >= 2100 || generation_time.year() >= 2100 {
```

Comment:
This seems arbitrary and creates a Serenity Y2.1K problem.

Reply:
It is fairly arbitrary. Do you have a better suggestion? I think it's not a great idea to query the current time in that function and compare to that; that feels outside the scope of what a "validate_token" function should do; plus it could break depending on OS time configuration.

Reply:
I do not think this is a necessary step for validation, at the end of the day it gets sent to discord to be accepted or rejected, this would be a more cursory review to ensure the formatting is proper I think. I don't think we are concerned with someone crafting a token with a huge or small timestamp which can't be used anyway.